src/engine/structures.rs

```diff
@@ -38,6 +38,15 @@ impl<'a, T: HasName + HasUuid> IntoIterator for &'a mut Table<T> {
     }
 }
 
+impl<'a, T: HasName + HasUuid> IntoIterator for &'a Table<T> {
```

Comment:
The existing one is for `&mut`, this one is for just `&`, which we need to iterate when we don't have a mutable reference.